ambry-replication/src/main/java/com/github/ambry/replication/RemoteReplicaInfo.java

```diff
@@ -86,6 +90,50 @@ public RemoteReplicaInfo(ReplicaId replicaId, ReplicaId localReplicaId, Store lo
     initializeTokens(token);
     // ExchangeMetadataResponse is initially empty. It will be populated by replica threads during replication cycles.
     this.exchangeMetadataResponse = new ReplicaThread.ExchangeMetadataResponse(ServerErrorCode.No_Error);
+    this.replicationRetryCount = 0;
+    this.MAX_REPLICATION_RETRY_COUNT = 0;
+  }
+
+  public RemoteReplicaInfo(ReplicationConfig replicationConfig, ReplicaId replicaId, ReplicaId localReplicaId,
+      Store localStore, FindToken token, long tokenPersistIntervalInMs, Time time, Port port) {
+    this.replicaId = replicaId;
+    this.localReplicaId = localReplicaId;
+    this.totalBytesReadFromLocalStore = 0;
+    this.localStore = localStore;
+    this.time = time;
+    this.port = port;
+    this.tokenPersistIntervalInMs = tokenPersistIntervalInMs;
+    initializeTokens(token);
+    // ExchangeMetadataResponse is initially empty. It will be populated by replica threads during replication cycles.
+    this.exchangeMetadataResponse = new ReplicaThread.ExchangeMetadataResponse(ServerErrorCode.No_Error);
+    this.replicationRetryCount = 0;
+    this.MAX_REPLICATION_RETRY_COUNT = replicationConfig.maxReplicationRetryCount;
+  }
```

Comment:
nit: we can create a private constructor like 
```
  public RemoteReplicaInfo(int maxReplicationRetryCount, ReplicaId replicaId, ReplicaId localReplicaId,
      Store localStore, FindToken token, long tokenPersistIntervalInMs, Time time, Port port) {
    this.replicaId = replicaId;
    this.localReplicaId = localReplicaId;
    this.totalBytesReadFromLocalStore = 0;
    this.localStore = localStore;
    this.time = time;
    this.port = port;
    this.tokenPersistIntervalInMs = tokenPersistIntervalInMs;
    initializeTokens(token);
    // ExchangeMetadataResponse is initially empty. It will be populated by replica threads during replication cycles.
    this.exchangeMetadataResponse = new ReplicaThread.ExchangeMetadataResponse(ServerErrorCode.No_Error);
    this.replicationRetryCount = 0;
    this.MAX_REPLICATION_RETRY_COUNT = maxReplicationRetryCount;
}
```

and then these two constructor would become
```
  public RemoteReplicaInfo(ReplicationConfig replicationConfig, ReplicaId replicaId, ReplicaId localReplicaId,
      Store localStore, FindToken token, long tokenPersistIntervalInMs, Time time, Port port) {
    this(replicaConfig.maxReplicationRetryCount, replicaId, localReplicaId, localStore, token, tokenPersistInIntervalInMs, time, port);
}

  public RemoteReplicaInfo(ReplicaId replicaId, ReplicaId localReplicaId, Store localStore, FindToken token,
      long tokenPersistIntervalInMs, Time time, Port port) {
    this(0, replicaId, localReplicaId, localStore, token, tokenPersistInIntervalInMs, time, port);
}
```

This way, we would have one method for initializing all the fields.

Reply:
done